### PR TITLE
Quickswap

### PR DESCRIPTION
**Pull request by AlexSaplin**

### CODE REVIEW

**Review by YusupovAI — APPROVED** [review]

File: src/interfaces/external/INonfungibleUniversalPositionManager.sol [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+interface INonfungibleUniversalPositionManager is IERC721 {}
```
Comment (k1rill-fedoseev) [inline src/interfaces/external/INonfungibleUniversalPositionManager.sol:6]: Seems like not used anymore

File: src/libraries/QuickswapV3FeesCalculation.sol [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import "@quickswap/core/IAlgebraPool.sol";
+import "@uniswap/v3-core/contracts/libraries/FullMath.sol";
+import "../interfaces/external/quickswapv3/INonfungibleQuickswapPositionLoader.sol";
+
+/// @title Math library for computing fees for quickswap v3 positions
+library QuickswapV3FeesCalculation {
+    uint256 public constant Q128 = 2**128;
+
+    /// @notice Calculate Uniswap token fees for the position with a given nft
+    /// @param pool UniswapV3 pool
```
Comment (k1rill-fedoseev) [inline src/libraries/QuickswapV3FeesCalculation.sol:13]:
```suggestion
    /// @notice Calculate QuickswapV3 token fees for the position with a given nft
    /// @param pool QuickswapV3 pool
```

File: src/libraries/QuickswapV3FeesCalculation.sol [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    /// @notice Get fee growth inside position from the tickLower to tickUpper since the pool has been initialised
+    /// @param pool QuickswapV3 pool
+    /// @param tickLower QuickswapV3 lower tick
+    /// @param tickUpper QuickswapV3 upper tick
+    /// @param tickCurrent QuickswapV3 current tick
+    /// @param feeGrowthGlobal0X128 QuickswapV3 fees of token0 collected per unit of liquidity for the entire life of the pool
+    /// @param feeGrowthGlobal1X128 QuickswapV3 fees of token1 collected per unit of liquidity for the entire life of the pool
+    /// @return feeGrowthInside0X128 The all-time fee growth in token0, per unit of liquidity, inside the position's tick boundaries, feeGrowthInside1X128 The all-time fee growth in token1, per unit of liquidity, inside the position's tick boundaries
+    function _getQuickswapFeeGrowthInside(
+        address pool,
```
Comment (k1rill-fedoseev) [inline src/libraries/QuickswapV3FeesCalculation.sol:61]: Does `IAlgebraPool pool` also work?

File: src/libraries/UniswapV3FeesCalculation.sol [hunk 4]
```diff
@@ -58,20 +58,18 @@ library UniswapV3FeesCalculation {
     /// @param feeGrowthGlobal1X128 UniswapV3 fees of token1 collected per unit of liquidity for the entire life of the pool
     /// @return feeGrowthInside0X128 The all-time fee growth in token0, per unit of liquidity, inside the position's tick boundaries, feeGrowthInside1X128 The all-time fee growth in token1, per unit of liquidity, inside the position's tick boundaries
     function _getUniswapFeeGrowthInside(
-        IUniswapV3Pool pool,
```
Comment (k1rill-fedoseev) [inline src/libraries/UniswapV3FeesCalculation.sol:61]: Why `IUniswapV3Pool pool` does not work?

File: test/AbstractIntegration.t.sol [hunk 5]
```diff
@@ -414,48 +405,14 @@ abstract contract AbstractIntegrationTestForVault is Test, SetupContract, Abstra
...
-        address liquidator = getNextUserAddress();
-        deal(address(token), liquidator, 10000 * 10**18, true);
-        vm.startPrank(liquidator);
-        token.approve(address(vault), type(uint256).max);
-        vm.expectRevert(Vault.PositionHealthy.selector);
-        vault.liquidate(vaultId); // hence not liquidated
-        vm.stopPrank();
-    }
-
     function LiquidationThresholdChangedHenceLiquidated() public {
```
Comment (k1rill-fedoseev) [inline test/AbstractIntegration.t.sol:408]:
```suggestion
    function testLiquidationThresholdChangedHenceLiquidated() public {
```

**Review by k1rill-fedoseev — APPROVED** [review]

**Pull request merged**